File: src/main/java/org/briarproject/briar/desktop/chat/ConversationMessageHeaderComparator.java
package org.briarproject.briar.desktop.chat;

import org.briarproject.briar.api.conversation.ConversationMessageHeader;

import java.util.Comparator;

public class ConversationMessageHeaderComparator
        implements Comparator<ConversationMessageHeader> {

    @Override
    public int compare(ConversationMessageHeader h1,
                       ConversationMessageHeader h2) {
        return Long.compare(h1.getTimestamp(), h2.getTimestamp());
    }

}

File: src/main/kotlin/org/briarproject/briar/desktop/chat/ConversationMessageHeaderComparator.kt
package org.briarproject.briar.desktop.chat

import org.briarproject.briar.api.conversation.ConversationMessageHeader

class ConversationMessageHeaderComparator : Comparator<ConversationMessageHeader> {

    override fun compare(
        h1: ConversationMessageHeader,
        h2: ConversationMessageHeader
    ): Int {
        return h1.timestamp.compareTo(h2.timestamp)
    }
}
